CodeGeneration/Sources/generate-swiftsyntax/templates/swiftsyntax/RawSyntaxValidationFile.swift

```diff
@@ -0,0 +1,176 @@
+                    }
+                  }
+
+                  var fileAndLine: (StaticString, UInt) {
+                    switch self {
+                    case .expectedNonNil(expectedKind: _, file: let file, line: let line):
+                      return (file, line)
+                    case .kindMismatch(expectedKind: _, actualKind: _, file: let file, line: let line):
+                      return (file, line)
+                  }
+                  }
+                }
+                """#
+              )
+
+              DeclSyntax(
+                """
+                func verify<Node: RawSyntaxNodeProtocol>(_ raw: RawSyntax?, as _: Node.Type, file: StaticString = #file, line: UInt = #line) -> ValidationError? {
+                  guard let raw = raw else {
+                    return .expectedNonNil(expectedKind: Node.self, file: file, line: line)
+                  }
+                  guard Node.isKindOf(raw) else {
+                    return .kindMismatch(expectedKind: Node.self, actualKind: raw.kind, file: file, line: line)
+                  }
+                  return nil
+                }
+                """
+              )
+
+              DeclSyntax(
+                """
+                func verify<Node: RawSyntaxNodeProtocol>(_ raw: RawSyntax?, as _: Node?.Type, file: StaticString = #file, line: UInt = #line) -> ValidationError? {
+                  if raw != nil {
+                    return verify(raw, as: Node.self, file: file, line: line)
+                  }
+                  return nil
+                }
+                """
+              )
+
+              DeclSyntax(
+                #"""
+                 func assertNoError(_ nodeKind: SyntaxKind, _ index: Int, _ error: ValidationError?) {
+                   if let error = error {
+                     let (file, line) = error.fileAndLine
+                     assertionFailure("""
+                       Error validating child at index \(index) of \(nodeKind):
+                       \(error.description)
+                       """, file: file, line: line)
+                       _ = 1
+                  }
+                }
+                """#
+              )
+
+              DeclSyntax(
+                #"""
+                func assertAnyHasNoError(_ nodeKind: SyntaxKind, _ index: Int, _ errors: [ValidationError?]) {
+                  let nonNilErrors = errors.compactMap({ $0 })
+                  if nonNilErrors.count == errors.count, let firstError = nonNilErrors.first {
+                    let (file, line) = firstError.fileAndLine
+                    assertionFailure("""
+                      Error validating child at index \(index) of \(nodeKind):
+                      Node did not satisfy any node choice requirement.
+                      Validation failures:
+                      \(nonNilErrors.map({ "- \($0.description)" }).joined(separator: "\n"))
+                      """, file: file, line: line)
+                    _ = 1
```

Comment:
Actually you can remove these `_ = 1` lines. I added them at some point during development to set a breakpoint on them and just forgot to remove them. Doesn’t need to be this PR though.